[Feature] window-switcher: add `order` parameter to `<windowSwitcher>` to switch between MRU and opened

mbroemme opened the pull request:
## Description
Add a new configuration option to control the window switcher traversal order. MRU cycling is convenient for quick toggling, but some users - me as well - prefer a stable taskbar-like order. This change introduces a choice while keeping MRU as the default and aligns the visual highlight with actual focus/stacking to avoid surprises.

New switcher order:
- `order="focus"` (default): cycle by most-recently-used focus order.
- `order="age"`: cycle by window creation/taskbar order.

johanmalm commented:
IIRC the taskbar order isn't consistent. I can't remember why, but it's not labwc related. 

I might be wrong on this; memory might be failing me.

If you open two taskbars at different times (with different number of clients open) that should confirm it.

Consolatis commented on src/osd/osd.c:
```diff
@@ -71,23 +71,66 @@ static struct view *
+	bool forwards = dir == LAB_CYCLE_DIR_FORWARD;
+	if (rc.window_switcher.order == WINDOW_SWITCHER_ORDER_OPENED) {
+		/* Build a stable list of candidates in open (creation) order */
+		struct wl_array views; wl_array_init(&views);
```
Please use two lines here, we don't usually have two statements within the same line.

mbroemme commented:
> This PR currently has a problem that window order shown in OSD isn't consistent, because the stacking-ordered array of windows is passed to `osd_impl->create()`.

Oops, forgot to test it. @tokyo4j Should I fix it or wait for #3235 and #3236 finalization?

Consolatis commented:
That would make the config format more weird when offering other options in the future (outside of linear or history) but that might be acceptable (I don't think there are many more potential sort orders anyway). Thus `linear="yes|no"` would be fine by me but lets give others a chance to voice their opposition.

mbroemme commented:
@tokyo4j I've updated and rebased it on the latest master with #3236 merged. It's much cleaner now since I was able to remove the custom sorting. Please review and test - it's working well on my end.

tokyo4j commented on docs/labwc-config.5.scd:
```diff
@@ -363,6 +363,32 @@ this is for compatibility with Openbox.
+	  are predictable.
+
+	Default is "focus".
+
```
This looks too verbose. I suggest more simpler documentation like below that fits in one paragraph and aligns with other settings in `<windowSwitcher>` (I'm a non-native English speaker and feel free to modify it). Sorry for not saying this earlier.

```suggestion
	*order* [focus|age] The order in which windows are cycled. *focus* cycles by
	recent focus history, starting with the previously focused window. *age* cycles
	by creation/open order, a stable taskbar-style ordering that doesn’t change on
	focus. Default is *focus*.
```

tokyo4j commented on include/view.h:
```diff
@@ -174,6 +174,7 @@ struct view {
 
 	bool mapped;
 	bool been_mapped;
+	uint64_t creation_iid;
```
nit: I don't know what "iid" means, and I suggest `creation_id` instead. Perhaps it's just I'm too bad at English, though.

tokyo4j commented:
LGTM other than my comments above. Tested and worked well. Thanks!
